cpp/src/Slice/Parser.cpp

```diff
@@ -136,7 +136,7 @@ Slice::DefinitionContext::setSeenDefinition()
 bool
 Slice::DefinitionContext::hasMetadata(const string& directive) const
```

Comment:
This PR slightly changes the semantics of `hasMetadata`.

Before, we needed an _exact_ match, but now `hasMetadata` will return true even for partial matches.
For example, if you marked a class with `format:compact` and called `hasMetadata("format")` it would return false before,
but now it will return true.

This actually lets us simplify a handful of places where we had this exact kind of case (metadata with arguments).

Reply:
Looking at how this is implemented, shouldn't we check it is `== directive` or  starts with `directive:`

In other words `hasMetadata("format")` should return `false` for `[formating]` or other cases where the common prefix refers to a different metadata.